tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,838 @@
+def _exec_cmd_on_pod(cmd, pod_obj):
+    """
+    Exec oc cmd on pods with retry
+
+    Args:
+        cmd (str): Command
+        pod_obj (obj): Pod object
+
+    """
+    pod_obj.exec_cmd_on_pod(cmd)
+
+
+def insert_delay(mon_dep):
+    """
+    Inserts delay on a monitor
+
+    Args:
+        mon_dep (str): Name of a monitor deployment
+
+    """
+    logger.info(f"Updating initialDelaySeconds on deployment: {mon_dep}")
+    cmd = (
+        f"oc get deployment {mon_dep} -o yaml | "
+        f'sed "s/initialDelaySeconds: 10/initialDelaySeconds: 10000/g" | oc replace -f - '
+    )
+    logger.info(f"Executing {cmd}")
+    os.system(cmd)
+
+
+def update_mon_initial_delay():
+    """
+    Inserts delay on all monitors
+
+    """
+    mon_dep = get_deployments_having_label(
+        label=constants.MON_APP_LABEL,
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    )
+    mon_deployments = [OCS(**mon) for mon in mon_dep]
+    for mon in mon_deployments:
+        logger.info(f"Updating initialDelaySeconds on {mon.name} deployment")
+        insert_delay(mon_dep=mon.name)
+
+    logger.info("Sleeping for mons to get initialized")
+    time.sleep(60)
+    logger.info("Validating whether all mons reached running state")
+    mon_pods = get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+    for mon in mon_pods:
+        wait_for_resource_state(resource=mon, state=constants.STATUS_RUNNING)
+
+
+def corrupt_ceph_monitors():
+    """
+    Corrupts ceph monitors by deleting store.db file
+
+    """
+    mon_pods = get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+    for mon in mon_pods:
+        logger.info(f"Corrupting monitor: {mon.name}")
+        mon_id = mon.get().get("metadata").get("labels").get("ceph_daemon_id")
+        logger.info(
+            _exec_cmd_on_pod(
+                cmd=f"rm -rf /var/lib/ceph/mon/ceph-{mon_id}/store.db", pod_obj=mon
+            )
+        )
+        try:
+            wait_for_resource_state(resource=mon, state=constants.STATUS_CLBO)
+        except ResourceWrongStatusException:
+            if (
+                mon.ocp.get_resource(resource_name=mon.name, column="STATUS")
+                != constants.STATUS_CLBO
+            ):
+                logger.info(
+                    f"Re-spinning monitor: {mon.name} since it did not reach CLBO state"
+                )
+                mon.delete()
+    logging.info("Validating all the monitors are in CLBO state")
+    for mon in get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE):
+        wait_for_resource_state(resource=mon, state=constants.STATUS_CLBO)
+
+
+def archive_and_mute_ceph_warn():
+    """
+    Archives all ceph crashes and mute warnings
+
+    """
+    toolbox = pod.get_ceph_tools_pod()
+    toolbox.exec_ceph_cmd(ceph_cmd="ceph crash archive-all")
+    toolbox.exec_ceph_cmd(
+        ceph_cmd="ceph config set mon auth_allow_insecure_global_id_reclaim false"
```

Comment:
any reason why we are muting the warnings here? is this part of the procedure?

Reply:
Same as above, not muting. But I will try to see if this reclaim warn goes after restarting some pods.

Reply:
re-spinning client and mon pods removed this, so need no to specifically set